src/plugins/intel_cpu/src/nodes/reference.cpp

```diff
@@ -136,7 +136,11 @@ ov::TensorVector Reference::prepareOutputs() const {
         void *dstDataPtr = getChildEdgesAtPort(i)[0]->getMemory().getData();
         ov::Shape shape = ngraphOp->get_output_partial_shape(i).rank().get_length() == 0 ?
                 ov::Shape{} : getChildEdgesAtPort(i)[0]->getMemory().getStaticDims();
-        outputs.push_back(ov::Tensor(ngraphOp->get_output_element_type(i), shape, dstDataPtr));
+        if (dstDataPtr != nullptr) {
+            outputs.push_back(ov::Tensor(ngraphOp->get_output_element_type(i), shape, dstDataPtr));
+        } else {
+            outputs.push_back(ov::Tensor(ngraphOp->get_output_element_type(i), shape));
+        }
```

Comment:
Isn't it more reliable to check the shape on presence of zeroes? There can be other reasons (and not all of them are legit) `dstDataPtr` being `nullptr`, but only in the context of an empty tensor, it's ok to allocate tensors on the ngraph side. Also what about inputs?